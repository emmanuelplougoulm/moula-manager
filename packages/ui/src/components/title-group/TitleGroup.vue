<template>
  <div class="container">
    <h1>{{ title }}</h1>
    <h2 v-if="subtitle">
      {{ subtitle }}
    </h2>
    <h3 v-if="description">
      {{ description }}
    </h3>
    <AlertItem v-if="alertText" :text="alertText" class="alert-text" />
  </div>
</template>

<script setup lang="ts">
import AlertItem from './../alert/AlertItem.vue';

defineProps({
  title: {
    type: String,
    required: true
  },
  subtitle: {
    type: String,
    default: undefined
  },
  description: {
    type: String,
    default: undefined
  },
  alertText: {
    type: String,
    default: undefined
  }
});
</script>

<style scoped>
.container {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  margin-bottom: 2.25rem;
}

:is(h1, h2, h3) {
  margin: 0;
}

h1 {
  color: var(--neutral-10);
  line-height: 1.813rem;
}

h2 {
  color: var(--primary-text);
  line-height: 1.5rem;
}

h3 {
  color: var(--neutral-40);
  line-height: 1.5rem;
}

.alert-text {
  margin-top: 0.625rem;
}
</style>
