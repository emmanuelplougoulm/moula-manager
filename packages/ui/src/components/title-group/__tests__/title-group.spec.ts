import { mount } from '@vue/test-utils';
import TitleGroup from '../TitleGroup.vue';

describe('TitleGroup tests', () => {
  it('should display the component with its title, subtitle, description and alert item', () => {
    const wrapper = mount(TitleGroup, {
      props: {
        title: 'Nunc sit amet consectetur ',
        subtitle: 'Nullam pellentesque felis ac faucibus vestibulum',
        description:
          'Phasellus quis justo dui. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.',
        alertText:
          'Quisque non massa nunc. Ut facilisis tincidunt elit, et porta enim finibus vitae. Quisque quis faucibus nunc. Maecenas molestie nisi laoreet mi finibus imperdiet.'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
