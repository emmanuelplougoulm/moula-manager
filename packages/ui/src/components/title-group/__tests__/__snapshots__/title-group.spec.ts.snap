// Vitest Snapshot v1

exports[`TitleGroup tests > should display the component with its title, subtitle, description and alert item 1`] = `
"<div data-v-1691f798=\\"\\" class=\\"container\\">
  <h1 data-v-1691f798=\\"\\">Nunc sit amet consectetur </h1>
  <h2 data-v-1691f798=\\"\\">Nullam pellentesque felis ac faucibus vestibulum</h2>
  <h3 data-v-1691f798=\\"\\">Phasellus quis justo dui. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.</h3>
  <p data-v-a2a80cda=\\"\\" data-v-1691f798=\\"\\" class=\\"text-default-regular alert-text\\">Quisque non massa nunc. Ut facilisis tincidunt elit, et porta enim finibus vitae. Quisque quis faucibus nunc. Maecenas molestie nisi laoreet mi finibus imperdiet.</p>
</div>"
`;
